datafusion/core/src/config.rs

```diff
@@ -102,6 +105,20 @@ impl ConfigDefinition {
             ScalarValue::UInt64(Some(default_value)),
         )
     }
+
+    /// Create a configuration option definition with a string value
+    pub fn new_string(
+        key: impl Into<String>,
+        description: impl Into<String>,
+        default_value: String,
+    ) -> Self {
+        Self::new(
+            key,
+            description,
+            DataType::UInt64,
```

Comment:
```suggestion
            DataType::Utf8,
```
I think the data type should be UTF8

Reply:
cc @waitingkuo

Reply:
LGTM just this minor comments